### PR TITLE
507 reevaluate `variable_types`

### PR DESCRIPTION
**Pull request by chlebowa**
companion to https://github.com/insightsengineering/teal.slice/pull/508

### CODE REVIEW

**Comment by chlebowa** [comment]
`variable_types` is used with `teal.slice:::variable_types`. Perhaps we should create a copy in this package's namespace like we do in `teal`?
`variable_types <- getFromNamespace("variable_types", ns = "teal.slice")` in `zzz.R`?

**Comment by averissimo** [comment]
> Perhaps we should create a copy in this package's namespace like we do in `teal`?
> `variable_types <- getFromNamespace("variable_types", ns = "teal.slice")` in `zzz.R`?

That seems reasonable and gets us ahead of tmc pre-release activities

**Comment by chlebowa** [comment]
Well, removing `teal.slice:::` causes a check warning:
```
❯ checking dependencies in R code ... NOTE
  Namespace in Imports field not imported from: ‘teal.slice’
    All declared Imports should be used.
```
Apparently this was the only thing used from `teal.slice`.

Now, `tmg` `Depends` on `teal` and `teal` `Depends` on `teal.slice`, so we wouldn't have to keep `teal.slice` in `tmg`'s `Imports` _BUT_ the minimal versions don't match:
`teal` `Depends` on `teal.slice >= 0.4.0.9027` and 
`tmg` `Depends` on `teal.slice >= 0.4.0.9028`.

I am for leaving `:::`.

**Comment by chlebowa** [comment]
This line just above implies that `df` is a `data.frame`-like, so it should be ok to use `vapply`.
`labels <- teal.data::col_labels(df)`

File: R/tm_variable_browser.R [hunk 1]
```diff
@@ -1189,7 +1189,7 @@ render_tab_table <- function(dataset_name, parent_dataname, output, data, input,
       )
 
       # get icons proper for the data types
-      icons <- stats::setNames(teal.slice:::variable_types(df), colnames(df))
+      icons <- teal.slice:::variable_types(df)
```
Comment (averissimo) [inline R/tm_variable_browser.R:1192]:
```suggestion
      icons <- vapply(df, function(x) class(x)[1L], character(1L))
```

**Comment by averissimo** [comment]
Is it too ambitions to remove `teal.slice` from `Depends`? as there's no longer a reason for it.

````diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index 98d2f0db..9c74dc7a 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -37,7 +37,6 @@ repos:
           - insightsengineering/teal.data
           - insightsengineering/teal.logger
           - insightsengineering/teal.reporter
-          - insightsengineering/teal.slice
           - insightsengineering/teal.transform
           - insightsengineering/teal.widgets
           - tern
diff --git a/DESCRIPTION b/DESCRIPTION
index 23c8b048..7ddf1031 100644
--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -40,7 +40,6 @@ Imports:
     teal.code (>= 0.4.1.9009),
     teal.logger (>= 0.1.1),
     teal.reporter (>= 0.2.0),
-    teal.slice (>= 0.4.0.9028),
     teal.widgets (>= 0.4.0),
     tern (>= 0.7.10),
     tibble (>= 2.0.0),
@@ -75,7 +74,7 @@ Config/Needs/verdepcheck: haleyjeppson/ggmosaic, tidyverse/ggplot2,
     daroczig/logger, tidyverse/magrittr, r-lib/scales, daattali/shinyjs,
     rstudio/shinyvalidate, dreamRs/shinyWidgets, tidyverse/stringr,
     insightsengineering/teal.code, insightsengineering/teal.logger,
-    insightsengineering/teal.reporter, insightsengineering/teal.slice,
+    insightsengineering/teal.reporter, 
     insightsengineering/teal.transform, insightsengineering/teal.widgets,
     insightsengineering/tern, tidyverse/tibble, tidyverse/tidyr,
     r-lib/tidyselect, tidymodels/broom, daattali/colourpicker,
diff --git a/staged_dependencies.yaml b/staged_dependencies.yaml
index 16fdd83d..1505e232 100644
--- a/staged_dependencies.yaml
+++ b/staged_dependencies.yaml
@@ -20,9 +20,6 @@ upstream_repos:
   insightsengineering/teal.reporter:
     repo: insightsengineering/teal.reporter
     host: https://github.com
-  insightsengineering/teal.slice:
-    repo: insightsengineering/teal.slice
-    host: https://github.com
   insightsengineering/teal.transform:
     repo: insightsengineering/teal.transform
     host: https://github.com
````

**Comment by chlebowa** [comment]
Thank you for maintaining focus in this :+1: 

I believe you're right, we can drop this dependency. @gogonzo please confirm.

**Comment by gogonzo** [comment]
> Thank you for maintaining focus in this 👍
> 
> I believe you're right, we can drop this dependency. @gogonzo please confirm.

Yup, `teal.slice` can be removed from DESCRIPTION file

**Comment by chlebowa** [comment]
This PR no longer depends on https://github.com/insightsengineering/teal.slice/pull/508 , so opening for review.

**Review by averissimo — APPROVED** [review]
Looks good to me, but I'm biased :sweat_smile:

**Pull request merged**